File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        String dataStr = String.valueOf(x);
        int left = 0;
        int right = dataStr.length() - 1;

        while (left < right) {

            if (dataStr.charAt(left) != dataStr.charAt(right)) {
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.rs
impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let conversion : String = x.to_string();

        let mut left : usize = 0;
        let mut right : usize = conversion.len() - 1;

        while left < right {
            if conversion.chars().nth(left).unwrap() != conversion.chars().nth(right).unwrap() {
                return false;
            }

            left += 1;
            right -= 1;
        }

        true
    }
}